docarray/array/array/io.py

```diff
@@ -302,6 +304,56 @@ def to_json(self) -> str:
         """
         return json.dumps([doc.json() for doc in self])
 
+    @classmethod
+    def _check_for_valid_document_type(cls) -> None:
```

Comment:
I would prefer returning a bool and rasing at the call site. That way it is clearer where the error actually occurs, and this method can be re-used in contexts where only the information is needed, without wanting to raise an exception